[trustar-150]: TruSTAR v2

hernantain opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

fixes: https://github.com/demisto/etc/issues/24555

## Description
New Version of TruSTAR integration. 
Rewriting the code from v1, adding tests, docstrings on code functions, adding new commands. 

For commands returning indicators we are putting data on 3 contexts: 
- The standard one without the malicious field as TruSTAR doesn't have a score for every indicator at this moment.
- DBotScore context with score as 0 for the same reason.
- TruSTAR context with all the information returned by the command.

## Screenshots
Indicator outputs on DBot, Standard and TruSTAR context

![476BDB0C-044D-4FB0-BC1B-86BA25E3BB69_1_105_c](https://user-images.githubusercontent.com/18565325/84542607-30cae600-acd0-11ea-9383-0718c7e16d41.jpeg)

## Minimum version of Demisto
- [ ] 4.5.0
- [x] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No: Is a new integration.

## Must have
- [x] Tests
- [x] Documentation 

## Demisto Partner?
- [x] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

Itay4 commented on Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml:
no need for this to be here, only the full context path is needed

Itay4 commented on Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml:
please add output types here

Itay4 commented on Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml:
what is the optional range here ? -1 to 3? please mention it in the description

Itay4 commented on Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml:
what is the optional range here ? -1 to 3? please mention it in the description

Itay4 commented on Packs/TruSTAR_V2/Integrations/TruSTAR_V2/TruSTAR_V2.yml:
what is the optional range here ? -1 to 3? please mention it in the description

hernantain commented:
@Itay4 I've:

- Corrected all small changes, typos, descriptions, parameters, etc.
- Moved Integration to existring TruSTAR pack
- Deprecating v1
- Implemented test-module
- Added test playbook

About the inconsistency with correlation_count on the different commands, we'll be solving the bug from our side. Same with domain/URL types. The `search-indicators` and `indicators-metadata` return the same TS object but the application differs within those 2, actually they can easily be linked on a playbook. And last, we are scoring DBot Indicators as Unknown as most of our indicators do not inform a malicious score. We prefer to change that once we can do it the same way for all of them.

hernantain commented on Packs/TruSTAR/Integrations/TruSTAR_V2/TruSTAR_V2.py:
```diff
@@ -0,0 +1,937 @@
+
+
+    def test_module(self):
+        response = self.client.ping()
```
It justs checks connectivity, the API does not verify the api key but of course that if the credentials are not OK, this command is going to fail.

Itay4 commented:
please revert the removal of https://github.com/demisto/content/blob/360918b7019aa5018b8478053883630600331f57/Pipfile

Itay4 commented on Packs/TruSTAR/Integrations/TruSTAR_V2/README.md:
```diff
@@ -0,0 +1,574 @@
+Use the TruSTAR v2 integration to manage reports, indicators and phishing submissions.
```
The README seem to be copy and pasted from the previous integration, and it is not according to our standard
please regenerate as described in https://xsoar.pan.dev/docs/integrations/integration-docs

Itay4 commented:
is this the expected behavior?
![image](https://user-images.githubusercontent.com/31018228/86224517-a28ca600-bb91-11ea-8784-20ebf7a20b91.png)

Itay4 commented on Packs/TruSTAR/Integrations/TruSTAR_V2/README.md:
```diff
@@ -0,0 +1,839 @@
+TruSTAR is an Intelligence Management Platform that helps you operationalize data across tools and teams, helping you prioritize investigations and accelerate incident response.
+This integration was integrated and tested with version xx of TruSTAR v2
```
can we set the `xx` version to something else?

Itay4 commented on Packs/TruSTAR/Integrations/TruSTAR_V2/README.md:
```diff
@@ -0,0 +1,839 @@
+| DBotScore.Score | number | The actual score | 
+
+
+#### Command Example
```
please add examples for all commands of command executions and outputs
